proto/codec/codec.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package codec;
```

Comment:
Let's use versioning in the package path like Temporal does: https://github.com/temporalio/api/tree/master/temporal/api

Reply:
@jlegrone - what do you think about adding the Python related code? If possible I'd like to defer merging the Python code. Maybe just having the proto would be enough?